ArchiSteamFarm/ArchiWebHandler.cs

```diff
@@ -1137,13 +1137,35 @@ internal sealed class ArchiWebHandler : IDisposable {
 				return null;
 			}
 
-			const string request = "/account/validatewalletcode";
+			const string requestValidateCode = "/account/validatewalletcode";
 
 			// Extra entry for sessionID
 			Dictionary<string, string> data = new Dictionary<string, string>(2) { { "wallet_code", key } };
 
-			Steam.RedeemWalletResponse response = await UrlPostToJsonObjectWithSession<Steam.RedeemWalletResponse>(SteamStoreURL, request, data).ConfigureAwait(false);
-			return response == null ? ((EResult Result, EPurchaseResultDetail? PurchaseResult)?) null : (response.Result, response.PurchaseResultDetail);
+			Steam.RedeemWalletResponse responseValidateCode = await UrlPostToJsonObjectWithSession<Steam.RedeemWalletResponse>(SteamStoreURL, requestValidateCode, data).ConfigureAwait(false);
+			if (responseValidateCode == null) {
+				return null;
+			}
+
+			if (responseValidateCode.Result != EResult.OK) {
+				return (responseValidateCode.Result, responseValidateCode.PurchaseResultDetail);
+			}
+
+			if (responseValidateCode.WalletCurrencyCode != responseValidateCode.RedeemingCodeWallet?.CurrencyCode) {
```

Comment:
There should be a null check above failing entire function, we do not expect to not have a wallet here (even if it can be `null` in the response, this should be probably logged as ASF null error, unless you can find an expected situation where you have `EResult.OK` and `null` wallet while ensuring that wallet is not null before in `Bot`.